common/persistence/persistenceInterface.go

```diff
@@ -70,6 +70,7 @@ type (
 		//  - number of rows actually deleted, if limit is honored
 		//  - UnknownNumRowsDeleted, when all rows below value are deleted
 		CompleteTasksLessThan(ctx context.Context, request *CompleteTasksLessThanRequest) (int, error)
+		GetOrphanTasks(ctx context.Context, request *GetOrphanTasksRequest) (*GetOrphanTasksResponse, error)
```

Comment:
Can you add a comment here. It's better to add a definition of "Orphan", from both user and technical point of view. 
I am still reading the code(thanks for your detailed PR description!), seems like for users, "Orphan task" means the task is completed or expired/timeout? Maybe nice to add a conclusion about technical definition of it.

Reply:
An orphan task would be a task record in the task table in the database for which no task list record exists. It no longer represents a valid item that can be acted upon. I can add a comment to that effect.

Reply:
sounds good to me.